### PR TITLE
Allow business to view date and time for completed surveys

### PR DESCRIPTION
**Pull request by shahi645**
# Motivation and Context
The business wanted to display the date and time next to a completed survey in the response status pop up. This would reduce the number of calls given to a respondent to get times. 

# What has changed
Able to view response status when complete, and then see date timestamp.

# How to test?
Set a case event to complete of completed by phone and check the response status

# Links
[Trello Card](https://trello.com/c/mr2EipGk/287-us208-int-display-date-and-time-stamp-alongside-rus-ce-response-status-m)

### CODE REVIEW

File: response_operations_ui/views/case.py [hunk 1]
```diff
@@ -31,18 +34,27 @@ def get_response_statuses(ru_ref, error=None):
 
     case_groups = case_controller.get_case_groups_by_business_party_id(reporting_unit['id'])
     case_group = case_controller.get_case_group_by_collection_exercise(case_groups, exercise['id'])
-
-    return render_template('change-response-status.html',
+    case_group_status = case_group['caseGroupStatus']
+    case_id = get_case_by_case_group_id(case_group['id']).get('id')
+    is_complete = False
```
Comment (andrewmil) [inline response_operations_ui/views/case.py]: We could create the `is_complete` and `completed_timestamp` objects in a slightly more functional way
```
is_complete = case_group_status in COMPLETE_STATE
completed_timestamp = get_timestamp_for_completed_case_event(case_id) if is_complete else None
```
Then you don't need to initialise them and then change them

File: response_operations_ui/controllers/case_controller.py [hunk 2]
```diff
@@ -193,9 +193,12 @@ def get_iac_count_for_case(case_id):
     return iac_count
 
 
-def get_case_events_by_case_id(case_id):
+def get_case_events_by_case_id_and_optional_category(case_id, categories=None):
```
Comment (andrewmil) [inline response_operations_ui/controllers/case_controller.py]: I'd probably leave the name of this as `get_case_events_by_case_id`, don't think we need to be so specific on what it's doing

File: response_operations_ui/templates/reporting-unit.html [hunk 3]
```diff
@@ -78,8 +78,10 @@ <h4 class="neptune">Collection exercises</h4>
               </td>
               <td name="tbl-ce-status" class="table--cell">
                 {{ ce.responseStatus }}
-                {% if ce.statuses|length > 0 and (ce.responseStatus == 'Not started' or ce.responseStatus == 'In progress')  %}
-                &nbsp;<a href="{{ url_for('case_bp.get_response_statuses', ru_ref=ru.sampleUnitRef, survey=survey.shortName, period=ce.exerciseRef) }}">Change</a>
+                {% if ce.responseStatus in ['Not started', 'In progress']  %}
```
Comment (andrewmil) [inline response_operations_ui/templates/reporting-unit.html]: Should this logic be taken out of the template?

File: response_operations_ui/templates/reporting-unit.html [hunk 4]
```diff
@@ -78,8 +78,10 @@ <h4 class="neptune">Collection exercises</h4>
...
+              &nbsp;   <a href="{{ url_for('case_bp.get_response_statuses', ru_ref=ru.sampleUnitRef, survey=survey.shortName, period=ce.exerciseRef, case_id=survey.case.id) }}">Change</a>
```
Comment (andrewmil) [inline response_operations_ui/templates/reporting-unit.html]: Formatting here looks a bit off here 🎨

File: response_operations_ui/views/case.py [hunk 5]
```diff
@@ -61,3 +73,11 @@ def update_response_status(ru_ref):
 
     return redirect(url_for('reporting_unit_bp.view_reporting_unit', ru_ref=ru_ref,
                     survey=short_name, period=period))
+
+
+def get_timestamp_for_completed_case_event(case_id):
+    complete_categories = ['OFFLINE_RESPONSE_PROCESSED', 'SUCCESSFUL_RESPONSE_UPLOAD', 'COMPLETED_BY_PHONE']
```
Comment (andrewmil) [inline response_operations_ui/views/case.py]: For consistency this should be defined at the top of the file or `COMPLETE_STATE` should be defined in it's function

File: response_operations_ui/templates/reporting-unit.html (hunk 3 above)
Comment (shahi645) [reply response_operations_ui/templates/reporting-unit.html]: I think in this instance its ok here i consulted Owen about this he said its fine. Plus to get the logic into the reporting units build up would add alot there to an already convoluted process.

File: response_operations_ui/templates/response-status.html [hunk 6]
```diff
@@ -55,25 +59,39 @@ <h1 class="jupiter">Change response status</h1>
...
+                                          {% for event, status in statuses.items() %}
+                                          <div class="field__item js-focusable-box">
+                                              <input required class="input input--radio js-focusable" name="event"
+                                                     value="{{ event }}" id="state-{{loop.index}}" type="radio">
+                                              <label class="label label--inline venus " for="state-{{loop.index}}">{{ status }}</label>
+                                          </div>
+                                          {% endfor %}
+                                      </fieldset>
+                                  </div>
+                                  <button type="submit" id="response-status-change-confirm-button" class="btn btn--primary u-mt-m">Confirm</button>
```
Comment (ryangrundy7) [inline response_operations_ui/templates/response-status.html]: In this part of the html page, the spacing looks off

**Pull request merged**